src/main/scala/com.snowplowanalytics.iglu/client/repositories/EmbeddedRepositoryRef.scala

```diff
@@ -134,21 +130,27 @@ case class EmbeddedRepositoryRef(override val config: RepositoryRefConfig, path:
    *         on Failure
    */
   // TODO: would be nice to abstract out failure.toProcessingMessage, and scrubbing
-  def lookupSchema(schemaKey: SchemaKey): Validated[Option[JsonNode]] = {
+  def lookupSchema(schemaKey: SchemaKey): ValidatedType[Option[JsonNode]] = {
     val schemaPath = s"${path}/schemas/${schemaKey.toPath}"
+    val streamOpt =
+      Option(EmbeddedRepositoryRef.getClass.getResource(schemaPath)).map(_.openStream())
+
     try {
-      JsonLoader.fromResource(schemaPath).some.success
+      streamOpt
+        .map(stream => new ObjectMapper().readTree(stream).asRight[ProcessingMessage]) // TODO: sequence on Validated doesn't work
+        .sequence
+        .toValidated
```

Comment:
shouldn't this be a nel?

Reply:
Actually @chuwy has a good point that a lot of these `Validated` data types should be just `Eithers`. I would rather change it in #105 though